examples/linear_deformation.py

```diff
@@ -0,0 +1,1187 @@
+            # Make a product space with one component
+            par_space = odl.ProductSpace(par_space, 1)
+        elif isinstance(par_space, odl.ProductSpace):
+            pass
+        else:
+            raise TypeError('expected Rn or ProductSpace as par_space, got '
+                            '{!r}.'.format(par_space))
+
+        if par_space.size != discr_space.ndim:
+            raise ValueError('dimensions of product space and image grid space'
+                             ' do not match ({} != {})'
+                             ''.format(par_space.size, discr_space.ndim))
+
+        # The operator maps from the parameter space to an inverse
+        # displacement for the domain of the target.
+
+        self.discr_space = discr_space
+        self.range_space = odl.ProductSpace(self.discr_space,
+                                            self.discr_space.ndim)
+
+        super().__init__(par_space, self.range_space, linear=True)
+
+        self.ft_kernel = ft_kernel
+
+        if not isinstance(control_points, odl.TensorGrid):
+            self._control_pts = np.asarray(control_points)
+            if self._control_pts.shape != (self.num_contr_pts, self.ndim):
+                raise ValueError(
+                    'expected control point array of shape {}, got {}.'
+                    ''.format((self.num_contr_pts, self.ndim),
+                              self.control_points.shape))
+        else:
+            self._control_pts = control_points
+
+        # TODO: check that number of control points is the same as alphas
+
+    @property
+    def ndim(self):
+        """Number of dimensions of the deformation."""
+        return self.domain.size
+
+    @property
+    def contr_pts_is_grid(self):
+        """`True` if the control points are given as a grid."""
+        return isinstance(self.control_points, odl.TensorGrid)
+
+    @property
+    def control_points(self):
+        """The control points for the deformations."""
+        return self._control_pts
+
+    @property
+    def num_contr_pts(self):
+        """Number of control points for the deformations."""
+        if self.contr_pts_is_grid:
+            return self.control_points.size
+        else:
+            return len(self.control_points)
+
+    @property
+    def image_grid(self):
+        """Spatial sampling grid of the image space."""
+        return discr_space.grid
+
+    def displacement_ft(self, alphas):
+        """Calculate the inverse translation at point y by n-D FFT.
+
+        inverse displacement: v(y)
+
+        Parameters
+        ----------
+        alphas: `ProductSpaceVector`
+            Deformation parameters for control points. It has ``n``
+            components, each of which has size ``N``. Here, ``n`` is
+            the number of dimensions and ``N`` the number of control
+            points. Note that here N = M.
+        kernel: Gaussian kernel fuction
+        """
+        ft_momenta = vectorial_ft_fit_op(alphas)
```

Comment:
`vectorial_ft_fit_op` is not defined at this point. It happens to be defined in this example since you happen to name an operator this way further down. As soon as you move this code out of this file, it will stop to work. What needs to be done here is that this vectorial FT operator must be defined *inside* this class, at least if we want to keep exactly this structure.

In fact, the computation of the vector field should not happen inside this operator, but rather in a new operator `KernelInterpolation` or similar. That's a better distribution of responsibilities: that new operator calculates the vector field (by interpolation in an RKHS), and this operator simply takes the vector field and "applies" it.